src/tooltip.js

```diff
@@ -102,7 +102,16 @@ class Tooltip extends Component {
 
   componentWillReceiveProps(nextProps) {
     const willBeVisible = nextProps.isVisible;
-    const { isVisible } = this.props;
+    const nextContent = nextProps.content;
+    const { isVisible, content} = this.props;
+
+    if (nextContent !== content && willBeVisible) {
+        // The location of the child element may have changed based on
+        // transition animations in the corresponding view, so remeasure
+        InteractionManager.runAfterInteractions(() => {
+          this.measureChildRect();
+        });
+    }
 
     if (willBeVisible !== isVisible) {
```

Comment:
Correct me if I'm wrong, but I think my preference would be to make this an `else if` statement like so

```
else if (willBeVisible !== isVisible) {
```

That way in the off-chance both conditions are true we don't go through both if-blocks

Reply:
Yes that's true